### DISCUSSION TITLE
关于MRR评价指标

### DISCUSSION THREAD

**Discussion by kuzma-long** [RUCAIBox/RecBole#1268] [Q&A] [answered]
我注意到很多论文里用MRR指标的时候都没有加@k，而recbole里的MRR指标都是加@k的，想知道一下这两种方式的区别，如果想不加的话如何操作呢？谢谢！

**Comment by Ethan-TZ** [comment, marked as answer]
@kuzma-long 您好，感谢您的关注！MRR指平均倒数排名，即计算所有用户推荐列表中第一个相关物品排名倒数的平均值，公式为：
![image](https://user-images.githubusercontent.com/69251138/165253369-661f09e8-1e56-4e1e-9131-c1e95cb9f8d7.png)
MRR用于召回任务，所以加@k取算法评分最高的k个物品(topk)作召回，所以一般情况下是加@k，而且RecBole只支持加@k的情况。而不加@k可能是更加关心第一个相关物品相对于所有物品打分中的排名情况，这可以等价于将k取一个较大的值。关于更多评测指标的细节，你可以参考[metrics](https://recbole.io/docs/recbole/recbole.evaluator.metrics.html)。
